receiver/memcachedreceiver/scraper.go

```diff
@@ -69,38 +65,133 @@ func (r *memcachedScraper) scrape(_ context.Context) (pdata.Metrics, error) {
 	ilm := md.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
 	ilm.InstrumentationLibrary().SetName("otelcol/memcached")
 
+	commandCount := initMetric(ilm.Metrics(), metadata.M.MemcachedCommands).Sum().DataPoints()
+	rUsage := initMetric(ilm.Metrics(), metadata.M.MemcachedRusage).Sum().DataPoints()
+	network := initMetric(ilm.Metrics(), metadata.M.MemcachedNetwork).Sum().DataPoints()
+	operationCount := initMetric(ilm.Metrics(), metadata.M.MemcachedOperations).Sum().DataPoints()
+	hitRatio := initMetric(ilm.Metrics(), metadata.M.MemcachedOperationHitRatio).Gauge().DataPoints()
+	bytes := initMetric(ilm.Metrics(), metadata.M.MemcachedBytes).Gauge().DataPoints()
+	currConn := initMetric(ilm.Metrics(), metadata.M.MemcachedCurrentConnections).Gauge().DataPoints()
+	totalConn := initMetric(ilm.Metrics(), metadata.M.MemcachedTotalConnections).Sum().DataPoints()
+	currItems := initMetric(ilm.Metrics(), metadata.M.MemcachedCurrentItems).Sum().DataPoints()
+	threads := initMetric(ilm.Metrics(), metadata.M.MemcachedThreads).Gauge().DataPoints()
+	evictions := initMetric(ilm.Metrics(), metadata.M.MemcachedEvictions).Sum().DataPoints()
+
 	for _, stats := range stats {
 		for k, v := range stats.Stats {
+			attributes := pdata.NewAttributeMap()
 			switch k {
 			case "bytes":
-				addIntGauge(ilm.Metrics(), metadata.M.MemcachedBytes.Init, now, parseInt(v))
+				addToIntMetric(bytes, attributes, parseInt(v), now)
```

Comment:
I think we need to fix the `parseInt` and `parseFloat` functions. We shouldn't be creating data points with a 0 value if parsing fails. There's a simple pattern established [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/httpdreceiver/scraper.go#L100:L132). Longer term, we will need to use `scrapererror` as well, but let's at least make this improvement so that the data is correct.

Reply:
Updated functions to better support errors